src/Resolvers/Microsoft.DotNet.MSBuildSdkResolver/MSBuildSdkResolver.cs

```diff
@@ -190,6 +191,22 @@ private sealed class CachedState
                         minimumVSDefinedSDKVersion);
                 }
 
+                string dotnetExe = Path.Combine(dotnetRoot, Constants.DotNet + Constants.ExeSuffix);
+                if (File.Exists(dotnetExe))
+                {
+                    propertiesToAdd ??= new Dictionary<string, string?>();
+                    propertiesToAdd.Add("DOTNET_HOST_PATH", dotnetExe);
+                }
+
+                string? runtimeVersion = dotnetRoot != null ?
+                    GetMSbuildRuntimeVersion(resolverResult.ResolvedSdkDirectory, dotnetRoot) :
+                    null;
+                if (runtimeVersion != null)
+                {
+                    propertiesToAdd ??= new Dictionary<string, string?>();
+                    propertiesToAdd.Add("MSBUILD_NET_TASKHOST_RUNTIME_VERSION", runtimeVersion);
```

Comment:
(same)

Reply:
This one I'm way more flexible on because I expect it to be a value that the MSBuild engine itself would start using once we start implementing the NetCore TaskHost, so we are more fully in the driver's seat here.

Reply:
changed to `MSBuildNetTaskHostRuntimeVersion`

Reply:
Oops. The convention is to start with "SdkResolver...". Renamed to "SdkResolverMSBuildTaskHostRuntimeVersion" (removed "Net" part because it is too long)